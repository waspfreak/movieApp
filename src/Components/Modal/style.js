import styled from 'styled-components';

export const StyledMovieModal = styled.div`
	.modal {
		position: fixed;
		top: 0;
		bottom: 0;
		left: 0;
		right: 0;
		width: 100%;
		z-index: 999999;
		display: flex;
		align-items: center;
		justify-content: center;
		background-color: rgba(0, 0, 0, 0.9);
		animation-name: appear;
		animation-duration: 0.5s;
	}

	.modal-dialog {
		border-radius: 0;
		max-width: 100%;
		background: transparent;
		position: relative;
		margin: 0 20px;
		max-height: calc(100vh - 100px);
		text-align: left;
		display: flex;
		flex-direction: column;
		overflow: hidden;
		-webkit-animation-name: animatetop;
		-webkit-animation-duration: 0.4s;
		animation-name: slide-in;
		animation-duration: 0.5s;
	}

	.modal-header,
	.modal-footer {
		display: flex;
		align-items: center;
		color: white;
		padding: 1rem;
		font-size: xx-large;
	}

	.modal-header {
		justify-content: space-between;
	}

	.modal-footer {
		border-top: 1px solid rgba(239, 235, 231, 0.9);
		justify-content: flex-end;
	}

	.modal-close {
		cursor: pointer;
		padding: 1rem;
		margin: -1rem -1rem -1rem auto;
	}

	.modal-body {
		overflow: auto;
	}

	.modal-content {
		padding: 1rem;
	}

	@keyframes appear {
		from {
			opacity: 0;
		}
		to {
			opacity: 1;
		}
	}

	@keyframes slide-in {
		from {
			transform: translateY(-150px);
		}
		to {
			transform: translateY(0);
		}
	}
`;
