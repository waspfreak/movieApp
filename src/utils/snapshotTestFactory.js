import {cleanup, render} from '@testing-library/react';
import React from 'react';

/**
 *
 * Sets up snapshot tests in a standard way and avoids repeating boiler plate
 */
export function snapshotTestFactory(Component: any, properties: any) {
	/**
	 * Re-usable it block
	 */
	const itBlock = (props: any) => {
		it('should match snapshot', () => {
			const {container} = render(<Component {...props} />);
			expect(container).toMatchSnapshot();
		});
	};

	/**
	 * Re-usable describe block
	 */
	describe(`${Component.name}`, () => {
		/**
		 * Clean up any mock functions that might be in props
		 */
		afterEach(() => {
			jest.clearAllMocks();
			cleanup();
		});

		/**
		 * If its an array of props then iterate them into seperate it blocks
		 */
		if (Array.isArray(properties)) {
			properties.forEach((props) => {
				itBlock(props);
			});
		} else {
			itBlock(properties);
		}
	});
}
