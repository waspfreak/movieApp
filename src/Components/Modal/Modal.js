import React, {useEffect} from 'react';
import {StyledMovieModal} from './style';

/**
 * Modal UI component for user interaction
 */

export const Modal = ({isVisible = false, children, onClose}) => {
	useEffect(() => {
		document.addEventListener('keydown', keydownHandler);
		return () => document.removeEventListener('keydown', keydownHandler);
	});

	function keydownHandler({key}) {
		switch (key) {
			case 'Escape':
				onClose();
				break;
			default:
		}
	}
	return !isVisible ? null : (
		<StyledMovieModal>
			<div className='modal' onClick={onClose}>
				<div className='modal-dialog' onClick={(e) => e.stopPropagation()}>
					<div className='modal-header'>
						<span className='modal-close' onClick={onClose}>
							&times;
						</span>
					</div>
					<div className='modal-body'>
						<div className='modal-content'>{children}</div>
					</div>
				</div>
			</div>
		</StyledMovieModal>
	);
};

export default Modal;
