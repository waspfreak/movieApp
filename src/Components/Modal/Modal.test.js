import {snapshotTestFactory} from '../../utils/snapshotTestFactory';
//
import {Modal} from './Modal';
const props: Modal[] = [
	{
		isVisible: false,
		children: 'content for modal',
		onClose: true,
	},
];
snapshotTestFactory(Modal, props);
