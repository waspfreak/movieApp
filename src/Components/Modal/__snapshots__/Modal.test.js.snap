// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should match snapshot 1`] = `<div />`;
